awx/ui/src/components/WorkflowOutputNavigation/WorkflowOutputNavigation.js

```diff
@@ -0,0 +1,120 @@
+import { t } from '@lingui/macro';
+import {
+  Select,
+  SelectOption,
+  SelectGroup,
+  SelectVariant,
+  Chip,
+} from '@patternfly/react-core';
+import ChipGroup from 'components/ChipGroup';
+
+const JOB_URL_SEGMENT_MAP = {
+  job: 'playbook',
+  project_update: 'project',
+  system_job: 'management',
+  system: 'system_job',
+  inventory_update: 'inventory',
+  workflow_job: 'workflow',
+};
+
+function WorkflowOutputNavigation({ relatedJobs }) {
+  const { id } = useParams();
+  const parent = document.getElementById('jobOuputCard');
+
+  const relevantResults = relatedJobs.filter(
+    ({
+      job: jobId,
+      summary_fields: {
+        unified_job_template: { unified_job_type },
+      },
+    }) => jobId && `${jobId}` !== id && unified_job_type !== 'workflow_approval'
+  );
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [filterBy, setFilterBy] = useState();
+  const [sortedJobs, setSortedJobs] = useState(relevantResults);
+
+  const handleFilter = (v) => {
+    if (filterBy === v) {
+      setSortedJobs(relevantResults);
+      setFilterBy();
+    } else {
+      setFilterBy(v);
+      setSortedJobs(
+        relevantResults.filter(
+          (node) =>
+            node.summary_fields.job.status === v.toLowerCase() &&
+            `${node.job}` !== id
+        )
+      );
+    }
+  };
+  const numSuccessJobs = relevantResults.filter(
+    (node) => node.summary_fields.job.status === 'successful'
+  ).length;
+  const numFailedJobs = relevantResults.length - numSuccessJobs;
+  const getOptions = () => [
+    <SelectGroup label={t`Workflow Statuses`} key="status">
+      <SelectOption
+        description={t`Filter by failed jobs`}
+        key="failed"
+        value={t`Failed`}
+        itemCount={numFailedJobs}
+      />
+      <SelectOption
+        description={t`Filter by successful jobs`}
+        key="successful"
+        value={t`Successful`}
+        itemCount={numSuccessJobs}
+      />
+    </SelectGroup>,
+    <SelectGroup label={t`Workflow Nodes`} key="nodes">
+      {sortedJobs?.map((node) => (
+        <SelectOption
+          key={node.id}
+          to={`/jobs/${
+            JOB_URL_SEGMENT_MAP[
+              node.summary_fields.unified_job_template.unified_job_type
+            ]
+          }/${node.summary_fields.job?.id}/output`}
+          component={Link}
+        >
+          {node.summary_fields.unified_job_template.name}
+        </SelectOption>
+      ))}
+    </SelectGroup>,
+  ];
+
+  return (
+    <Select
+      key={`${id}`}
+      variant={SelectVariant.typeaheadMulti}
+      menuAppendTo={parent}
+      onToggle={() => {
+        setIsOpen(!isOpen);
+      }}
+      selections={filterBy}
+      onSelect={(e, v) => {
+        if (v !== 'Failed' && v !== 'Successful') return;
+        handleFilter(v);
+      }}
+      isOpen={isOpen}
+      isGrouped
+      hasInlineFilter
+      placeholderText={t`Workflow Job 1/${relevantResults.length}`}
```

Comment:
I went around several times with this, and I landed on this pattern because workflows are not always linear so how does someone tell if the output being reviewed is #3 in a workflow, or some other number?  I landed on `1/n` pattern simply as a way to tell the user you are looking a 1 node output, and there are `n` nodes total.